receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -177,32 +180,57 @@ func (prw *prometheusRemoteWriteReceiver) translateV2(_ context.Context, req *wr
 		// Instead of creating a whole new OTLP metric, we just append the new sample to the existing OTLP metric.
 		// This cache is called "intra" because in the future we'll have a "interRequestCache" to cache resourceAttributes
 		// between requests based on the metric "target_info".
-		intraRequestCache = make(map[uint64]pmetric.ResourceMetrics)
 		// The key is composed by: resource_hash:scope_name:scope_version:metric_name:unit:type
 		// TODO: use the appropriate hash function.
 		metricCache = make(map[string]pmetric.Metric)
 	)
 
 	for _, ts := range req.Timeseries {
 		ls := ts.ToLabels(&labelsBuilder, req.Symbols)
-		if !ls.Has(labels.MetricName) {
+		if !ls.Has(labels.MetricName) && ts.Metadata.Type != writev2.Metadata_METRIC_TYPE_INFO {
 			badRequestErrors = errors.Join(badRequestErrors, fmt.Errorf("missing metric name in labels"))
 			continue
 		} else if duplicateLabel, hasDuplicate := ls.HasDuplicateLabelNames(); hasDuplicate {
 			badRequestErrors = errors.Join(badRequestErrors, fmt.Errorf("duplicate label %q in labels", duplicateLabel))
 			continue
 		}
 
+		// If it is a metric of type INFO, we use its labels as attributes of the resource
+		// Ref: https://opentelemetry.io/docs/specs/otel/compatibility/prometheus_and_openmetrics/#resource-attributes-1
+		if ts.Metadata.Type == writev2.Metadata_METRIC_TYPE_INFO {
```

Comment:
It's not any metric of the kind `INFO`, just the metric with the name `target_info`. I completely understand the confusion here.

[The TYPE Info is mentioned by the exposition format OpenMetrics](https://prometheus.io/docs/specs/om/open_metrics_spec/#info), but in reality Prometheus has never adapted its storage to handle OM's Info metrics differently and therefore never pushes an Info metric via remote write.


We'll need to switch the logic here completely to only get resource attributes from the specific metric with name `target_info`